src/System.Linq/src/System/Linq/AppendPrepend.cs

```diff
@@ -523,6 +530,23 @@ public override int GetCount(bool onlyIfCheap)
 
                 return !onlyIfCheap || _source is ICollection<TSource> ? _source.Count() + (_appended == null ? 0 : _appended.Count) + (_prepended == null ? 0 : _prepended.Count) : -1;
             }
+
+            public override HashSet<TSource> ToHashSet(IEqualityComparer<TSource> comparer)
+            {
+                HashSet<TSource> hashSet = new HashSet<TSource>(_source, comparer);
```

Comment:
Again,  `_source.ToHashSet(comparer)` might take an optimised path. I won't continue pointing to such cases.